### DISCUSSION TITLE
Entangle and :class not working properly together (Playground Included)

### DISCUSSION THREAD

**Discussion by EricBusch** [livewire/livewire#2150] [1. Help]
Hi All,

I have a component which updates the CSS classes for specific elements. For some elements the classes are overwritten properly but for other elements the classes are appended, not overwritten.

Here's my playground: https://laravelplayground.com/#/snippets/c7b1f7ee-da6e-47e4-81d5-fc9961b30a2c

To generate the "error", follow these steps:

1. Check "2xl" option
1. Check "numbers" option
1. Check "xs" option

### Desired Output

> > <img width="352" alt="desired-output" src="https://user-images.githubusercontent.com/12106436/100920506-36e49700-34a9-11eb-95c8-e65535f6ed78.png">

### Actual Output

> > <img width="352" alt="incorrect-output" src="https://user-images.githubusercontent.com/12106436/100920521-3e0ba500-34a9-11eb-8fd0-0175c4efcde1.png">

### Inspector shows classes not being updated properly.

> > <img width="677" alt="multiple-classes" src="https://user-images.githubusercontent.com/12106436/100920562-4e238480-34a9-11eb-86b5-a0bba96d149c.png">

Any ideas why the last 2 elements are not having their classes updated properly?

Thanks
Eric

**Comment by joshhanley** [comment]
@EricBusch it looks like you are running into a morphDom issue. The reason is to do with your wire:key isn't unique in relation to the vocab set.

I've got it working here https://laravelplayground.com/#/snippets/891c82a6-6fc3-4c42-9de3-a8f97843263a

On line 24 I've added {{$vocabSet}} to tell Livewire that there is a difference between the numbers set and the colour set.

Be aware that there was also some bugs with entangle that have been fixed (in 2.3.5), but Livewire Playground is still running an old version of Livewire. So please test out in your local environment with the latest versions of Livewire and Alpine.

Hope this helps!

**Comment by EricBusch** [comment]
Thanks @joshhanley 

I noticed that you also removed `wire:model` from the font size radio `<input>` elements.

Does that mean when you are using Entangle that `wire:model` is redundant?

**Reply by joshhanley** [reply]
@EricBusch no worries! 

Oh yeah (missed that), you don't need both as entangle keeps Livewire and Alpine in sync so you can either user x-model or wire:model, but you don't need both.

**Reply by EricBusch** [reply]
Thanks for clarifying that. That's super useful!

I wonder if this is a bug with Livewire. The [playground that you created](https://laravelplayground.com/#/snippets/891c82a6-6fc3-4c42-9de3-a8f97843263a) does work but after you change the Vocab Set, then changing the Font Size makes a network request each time.

I even tested this on my own site after upgrading Alpine and still no luck.

Do you think I should file a bug?

**Reply by joshhanley** [reply]
@EricBusch you just need to add .defer to the end of entangle `fontSizeTerm0: @entangle('fontSizeTerm0').defer` and that will stop network requests for font size.

I think I might have removed it while I was testing, because looking at your original example it has it 🙂
